tools/ubuntu-setup/docker.sh

```diff
@@ -3,28 +3,32 @@ set -e
 set -x
 
 sudo apt-get -y install apt-transport-https ca-certificates
-sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 36A1D7869245C8950F966E92D8576A8BA88D21E9
-sudo sh -c "echo deb https://apt.dockerproject.org/repo ubuntu-trusty main  > /etc/apt/sources.list.d/docker.list"
-sudo apt-get -y update -qq
 
-sudo apt-get purge lxc-docker
-sudo apt-cache policy docker-engine
+# Docker GPG Key
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+
+sudo add-apt-repository \
+   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+   $(lsb_release -cs) \
+   stable"
+
+sudo apt-get -y update -qq
 
 # AUFS
-sudo apt-get -y install linux-image-extra-$(uname -r)
+sudo apt-get -y install linux-image-extra-$(uname -r) linux-image-extra-virtual
 
 # DOCKER
-sudo apt-get install -y --force-yes docker-engine=1.12.0-0~trusty
+sudo apt-get install -y --force-yes docker-ce
```

Comment:
I think we want to install docker 1.12 if possible since it's the version we support